### ISSUE TITLE
Regression with ellipsis handling in v1.9.0

### ISSUE THREAD

**Issue by aiudirog** [python/mypy#17019]
**Bug Report**

With v1.9.0, `...` no longer appears to be considered the only instance of the `EllipsisType` and this causes type narrowing to not function properly.

**To Reproduce**

```python
from types import EllipsisType
x: int | str | EllipsisType = ...
y: int | str = 0 if x is ... else x
```

https://mypy-play.net/?mypy=1.9.0&python=3.12&flags=strict&gist=02e73fd9b069629261e7328ab3530973

**Expected Behavior**

No errors, as it behaved on 1.8.0:

```python
Success: no issues found in 1 source file
```

**Actual Behavior**

```
main.py:3: error: Incompatible types in assignment (expression has type "int | str | EllipsisType", variable has type "int | str")  [assignment]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.9.0
- Mypy command-line flags: --strict
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: 3.10.13 & 3.12 (in the playground)

**Comment by JelleZijlstra** [comment]
Duplicate of #17002

**Comment by aiudirog** [comment]
My apologies, I must have only checked open tickets and didn't see the closed one
